src/librustc_mir/interpret/eval_context.rs

```diff
@@ -683,8 +694,8 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         }
         // done
         self.frame_mut().locals = locals;
-
         M::after_stack_push(self)?;
+        self.frame_mut().loc = Ok(mir::Location::START);
```

Comment:
It's about the order of these two lines: for errors that happen during `after_stack_push`, so we want to use the span of the first statement, or the span of the entire body? IMO the latter makes more sense, the first statement has nothing to do with this.

For the stack overflow, probably the best span to use would be that of the call site.

Reply:
yea, the body's span is good here, though it's actually the call site's fault. But lets leave such refactorings (using the call site span) to the future, I remember trying something similar a while back and it just complicated the span scheme even more

Reply:
Hm, this wouldn't even be that hard I think... we can just call `cur_span` before pushing the new frame, then we should get the call site.

Reply:
It's just not clear to me that we always want the call site. The callee could also be responsible for the problem -- in fact I'd usually think they are.

Reply:
at runtime the decision whose job it is to create the stack frame is an ABI decision, but that doesn't really make sense for CTFE. But I also think your alluding to a different thing? If an required_const will fail regardless of the concrete generic parameters, then that's indeed the function's fault and not the caller's

Reply:
> If an required_const will fail regardless of the concrete generic parameters, then that's indeed the function's fault and not the caller's

Yes, that is one possible scenario I had in mind where blaming the caller seems wrong.